compiler/luci/service/src/Nodes/CircleReshape.cpp

```diff
@@ -117,12 +117,32 @@ loco::TensorShape Algorithm::visit(const luci::CircleReshape *node)
     }
     else
     {
-      auto shape_node = loco::must_cast<luci::CircleNode *>(node->shape());
-      assert(shape_node->rank() == 1);
-      // shape_node tensor values will provide new shape, like [2, 3, 4]
-      auto num_elements = shape_node->dim(0).value(); // above example will give 3
-      shape_by_input.rank(num_elements);
-      is_static_shape = false;
+      auto new_shape = node->newShape();
+      auto num_ele = new_shape->rank();
+      if (num_ele)
+      {
+        is_static_shape = true;
+        shape_by_input.rank(num_ele);
+        for (uint32_t i = 0; i < num_ele; ++i)
+        {
+          if (new_shape->dim(i) > 0)
+            shape_by_input.dim(i) = new_shape->dim(static_cast<uint32_t>(i));
+          else
+          {
+            is_static_shape = false;
+            shape_by_input.dim(i).unset();
+          }
+        }
+      }
+      else
```

Comment:
The previous logic only runs when rank is 0? It seems that the previous logic is not for rank 0.

Reply:
Maybe we can do something like this?

If `shape` exists -> Do the previous logic
If `shape` does not exist, and `newShape` exists (!= nullptr) -> Do the current logic
Otherwise -> throw an exception

Does it solve your problem?

Reply:
I'm not sure about the priority how to use two: `shape` input vs `newShape` attribute from option. From the tflite codes, I assume two should be the same.
Existing codes are implemented from the assumption `same`

Current problem is from `shape` input cannot be determined as it's not `CircleConst` but `newShape` exist, where current code doesn't check `newShape`.

This PR is to do something about the `newShape` when `shape` cannot be determined.
When `newShape` exist, rank should be > 0, as default is 0, somewhere it was set.

If you have any other opinion, please share. I'd be glad to accept :)